src/main/java/com/zilen/weather/exception/CityNotFoundException.java

```diff
@@ -0,0 +1,19 @@
+package com.zilen.weather.exception;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+
+@ControllerAdvice
+public class CityNotFoundException extends RuntimeException {
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<Object> handleCityIsNullException(
```

Comment:
Here you create a global exception for all your controllers, please create your own exception and add handling only for city controller